lib/cisco_node_utils/vni.rb

```diff
@@ -91,7 +97,13 @@ def create
     end
 
     def destroy
-      config_set('vni', 'destroy', @vni_id)
+      if /N9K/.match(node.product_id)
+        # Just destroy the vni-vlan mapping
+        vlan = mapped_vlan
+        config_set('vni', 'mapped_vlan', vlan: vlan, state: 'no', vni: @vni_id)
+      else
+        config_set('vni', 'destroy', @vni_id)
+      end
```

Comment:
Didn't quite understand.

for N7K, we need "no vni <vni_id>"
for N9K, we need 
vlan <vlan_id>
  no vn-segment <vni_id>

Not sure how the method proposed by you (single config_get) will work for both the syntaxes.

I have gotten rid of vlan = mapped_vlan though.

    def destroy
      if /N9K/.match(node.product_id)
        # Just destroy the vni-vlan mapping
        config_set('vni', 'mapped_vlan', vlan: mapped_vlan, state: 'no', vni: @vni_id)
      else
        config_set('vni', 'destroy', vni: @vni_id)
      end
    end